lib/cli/interactive-setup/aws-credentials.js

```diff
@@ -86,33 +208,236 @@ const steps = {
+    const linkUid = getProviderLinkUid(app, service);
+    let orgUid;
+    try {
+      orgUid = await getOrgUidByName(orgName);
+    } catch (err) {
+      if (err.code === 'DASHBOARD_UNAVAILABLE') {
+        process.stdout.write(`\n${chalk.yellow(err.message)}\n`);
+        return false;
+      }
+      throw err;
+    }
+    try {
+      await sdk.createProviderLink(orgUid, linkType, linkUid, providerUid);
+      return true;
+    } catch (err) {
+      if (err.statusCode && err.statusCode >= 500) {
+        process.stdout.write(
+          `\n${chalk.yellow(
+            'Dashboard service is currently unavailable, please try again later'
+          )}\n`
+        );
+        return false;
+      }
+      throw err;
+    }
+  },
+};
+
+const isUserLoggedIn = () => {
+  if (process.env.SERVERLESS_ACCESS_KEY) {
+    return true;
+  }
+
+  if (configUtils.getLoggedInUser()) {
+    return true;
+  }
+
+  return false;
 };
 
 module.exports = {
-  async isApplicable({ configuration }) {
+  async isApplicable(context) {
+    const { configuration, history } = context;
     if (
       _.get(configuration, 'provider') !== 'aws' &&
       _.get(configuration, 'provider.name') !== 'aws'
     ) {
       return false;
     }
     if (new AWS.S3().config.credentials) return false;
-    return !(await awsCredentials.resolveFileProfiles()).size;
+    if ((await awsCredentials.resolveFileProfiles()).size) return false;
+
+    const orgName = configuration.org;
+    if (orgName && isUserLoggedIn()) {
+      let providers;
+      try {
+        providers = await getProviders(orgName);
+      } catch (err) {
+        if (err.code === 'DASHBOARD_UNAVAILABLE') {
+          process.stdout.write(`\n${chalk.yellow(err.message)}\n`);
+          return false;
+        }
+        throw err;
+      }
+      const hasDefaultProvider = providers.some((provider) => provider.isDefault);
+
+      if (hasDefaultProvider) return false;
+
+      // For situation where it is invoked for already existing service
+      // We need to check if service already has a linked provider
+      if (
+        providers &&
+        !history.has('service') &&
+        (await doesServiceHaveLinkedProvider(orgName, configuration.app, configuration.service))
+      ) {
+        return false;
+      }
+    }
+
+    return true;
   },
   async run(data) {
-    process.stdout.write(
-      'No AWS credentials were found on your computer, ' +
-        'you need these to host your application.\n\n'
-    );
-
     return module.exports.runSteps(data);
   },
   steps,
   runSteps: async (context) => {
-    if (!(await steps.shouldSetupAwsCredentials())) return;
-    await steps.ensureAwsAccount();
+    let providers;
 
-    await steps.ensureAwsCredentials(context);
-    await steps.inputAwsCredentials();
+    // It is possible that user decides to not configure org for his service and
+    // we still should allow setup of local credentials in such case
+    if (context.configuration.org) {
```

Comment:
I think here again we should check `isUserLoggedIn` (?)

Reply:
great catch, I missed that